pandas/computation/tests/test_eval.py

```diff
@@ -750,6 +750,28 @@ def check_chained_cmp_op(self, lhs, cmp1, mid, cmp2, rhs):
 
 ENGINES_PARSERS = list(product(_engines, expr._parsers))
 
+#-------------------------------------
+# typecasting rules consistency with python
+
+class TestTypeCasting(object):
+
+    def check_binop_typecasting(self, engine, parser, op, dt):
+        tm.skip_if_no_ne(engine)
+        s = 'df {} 3'.format(op)
+        df = mkdf(5, 3, data_gen_f=f, dtype=dt)
+        res = pd.eval(s, engine=engine, parser=parser)
+        assert df.values.dtype == dt
+        assert res.values.dtype == dt
+        assert_frame_equal(res, eval(s))
+
+    def test_binop_typecasting(self):
+        for engine, parser in ENGINES_PARSERS:
+            for op in ['+', '-', '*', '**', '/']:
+                # maybe someday... numexpr has too many upcasting rules now
+                #for dt in chain(*(np.sctypes[x] for x in ['uint', 'int', 'float'])):
+                for dt in [np.float32, np.float64]:
```

Comment:
numexpr does not support float16, and for ints numexpr was upcasting sum of two ints to int64 (I am guessing for overflow reasons).  Changing this behavior now could break other people's code (worse, it would only happen on some inputs), so I left it alone